src/mofaflex/_core/model.py

```diff
@@ -88,13 +112,20 @@ def __init__(
 
         self.scale_elbo = True
         n_views = len(self.view_names)
+        n_guiding_vars = len(self.guiding_vars_names)
         self.view_scales = dict.fromkeys(self.view_names, 1.0)
-        if self.scale_elbo and n_views > 1:
+        self.guiding_vars_scales = dict.fromkeys(self.guiding_vars_names, 1.0)
+        if self.scale_elbo and n_views + n_guiding_vars > 1:
             for view_name, view_n_features in n_features.items():
-                self.view_scales[view_name] = (n_views / (n_views - 1)) * (
-                    1.0 - view_n_features / sum(n_features.values())
+                self.view_scales[view_name] = ((n_views + n_guiding_vars) / ((n_views + n_guiding_vars) - 1)) * (
+                    1.0 - view_n_features / (sum(n_features.values()) + n_guiding_vars)
```

Comment:
Does that make sense? Guiding variables are not per view, a guiding variable applies to all views. This is meant to make the optimization procedure pay more attention to views with few features.